src/utils/download.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ProgressBar from 'progress';
+import { debugLogger } from './debugLogger';
+import { getErrorMessage, httpRequest } from './utils';
+import * as fs from 'fs';
+
+
+type OnProgressCallback = (downloadedBytes: number, totalBytes: number) => void;
+type DownloadFileLogger = (message: string) => void;
+
+function toMegabytes(bytes: number): string {
+  const mb = bytes / 1024 / 1024;
+  return `${Math.round(mb * 10) / 10} Mb`;
+}
+
+function getDownloadProgress(progressBarName: string): OnProgressCallback {
+  let progressBar: ProgressBar;
+  let lastDownloadedBytes = 0;
+
+  return (downloadedBytes: number, totalBytes: number) => {
+    if (!process.stderr.isTTY) return;
+    if (!progressBar) {
+      progressBar = new ProgressBar(
```

Comment:
nit: can we keep the original formatting? I think that changing formatting when moving the code just makes it much harder to follow what's going on.

Reply:
I think `eslint -> "curly": [2, "multi-or-nest", "consistent"]` was the culprit here. 
Changing to all might be helpful.